extensions/cocostudio/timeline/CCBoneNode.js

```diff
@@ -0,0 +1,561 @@
+                    miny = cc.rectGetMinY(r);
+                    maxx = cc.rectGetMaxX(r);
+                    maxy = cc.rectGetMaxY(r);
+
+                    first = false;
+                } else {
+                    minx = Math.min(cc.rectGetMinX(r), minx);
+                    miny = Math.min(cc.rectGetMinY(r), miny);
+                    maxx = Math.max(cc.rectGetMaxX(r), maxx);
+                    maxy = Math.max(cc.rectGetMaxY(r), maxy);
+                }
+                displayRect.setRect(minx, miny, maxx - minx, maxy - miny);
+            }
+            return displayRect;
+        },
+
+        getBoundingBox: function () {
+            var boundingBox = this.getVisibleSkinsRect();
+            return cc.rectApplyAffineTransform(boundingBox, this.getNodeToParentAffineTransform());
+        },
+
+        getBoneToSkeletonTransform: function () {
+            var retMat;
+            if (this._rootSkeleton == null) {
+                retMat = {a: 1, b: 0, c: 0, d: 1, x: 0, y: 0};
+                debug.log("can not transform before added to Skeleton");
+                return retMat;
+            }
+
+            retMat = this.getNodeToParentTransform();
+            //todo check here p exists
+            for (var p = this._parent; p && p != this._rootSkeleton; p = p.getParent()) {
+                //todo check here
+                retMat = cc.affineTransformConcat(p.getNodeToParentTransform(), retMat);
+            }
+            return retMat;
+        },
+
+        getBoneToSkeletonAffineTransform: function () {
+            var retTrans;
+            if (this._rootSkeleton == null) {
+                retTrans = {a: 1, b: 0, c: 0, d: 1, x: 0, y: 0};
+                debug.log("can not transform before added to Skeleton");
+                return retTrans;
+            }
+            retTrans = this.getNodeToParentAffineTransform();
+            for (var p = this._parent; p && p != this._rootSkeleton; p = p.getParent())
+                retTrans = cc.affineTransformConcat(retTrans, p.getNodeToParentAffineTransform());
+            return retTrans;
+        },
+
+        batchBoneDrawToSkeleton: function (bone) {},
+
+        setLocalZOrder: function (localZOrder) {
+            Node.prototype.setLocalZOrder.call(this, localZOrder);
+            if (this._rootSkeleton != null)
+                this._rootSkeleton._sortedAllBonesDirty = true;
+        },
+
+        setName: function (name) {
+            var oldName = this.getName();
+            Node.prototype.setName.call(this, name);
+            if (this._rootSkeleton != null) {
+                var item = this._rootSkeleton._subBonesMap[oldName];
+                if (item) {
+                    delete this._rootSkeleton._subBonesMap[oldName];
+                    this._rootSkeleton._subBonesMap[name] = item;
+                }
+            }
+        },
+
+        isPointOnRack: function (bonePoint) {
+            if (bonePoint.x >= 0 && bonePoint.y >= 0
```

Comment:
this func just for cocostudio,  can remove it in js,  see #ifdef CC_STUDIO_ENABLED_VIEW